src/textual/widgets/_data_table.py

```diff
@@ -1175,7 +1178,7 @@ def clear(self, columns: bool = False) -> None:
         self.hover_coordinate = Coordinate(0, 0)
         self._label_column = Column(self._label_column_key, Text(), auto_width=True)
         self._labelled_row_exists = False
-        self.refresh()
+        return self.refresh()
```

Comment:
I think it is better to explicitly return self, rather than return the result of `refresh`.

Otherwise we create a tight coupling between the two methods. i.e. if `refresh` changes, we break `clear`